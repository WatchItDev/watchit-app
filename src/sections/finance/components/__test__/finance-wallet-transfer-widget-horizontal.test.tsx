import { describe, expect, it } from "vitest";
import { render, screen } from "@testing-library/react";
import FinanceWalletTransferWidgetHorizontal from "../finance-wallet-transfer-widget-horizontal";

const mockProps = {
  icon: "icon-name",
  title: "Test Title",
  value: 100,
  color: "red",
};

const renderComponent = (props = mockProps) =>
  render(<FinanceWalletTransferWidgetHorizontal {...props} />);

describe("[COMPONENTS] <FinanceWalletTransferWidgetHorizontal/>", () => {
  it("to match snapshot", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it("should render with correct props", () => {
    renderComponent();
    expect(screen.getByText(mockProps.title)).toBeInTheDocument();
    expect(screen.getByText(`${mockProps.value} MMC`)).toBeInTheDocument();
  });
});
