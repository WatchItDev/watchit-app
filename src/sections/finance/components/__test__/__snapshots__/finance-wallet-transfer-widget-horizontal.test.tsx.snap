// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[COMPONENTS] <FinanceWalletTransferWidgetHorizontal/> > to match snapshot 1`] = `
<div>
  <div
    class="MuiStack-root css-1yxi6tj-MuiStack-root"
  >
    <div
      class="MuiStack-root css-19wq847-MuiStack-root"
    >
      <span
        data-icon="icon-name"
        data-sx="{"color":"red","position":"absolute"}"
        data-testid="icon"
        width="32"
      />
      <span
        aria-valuenow="100"
        class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary css-179juh6-MuiCircularProgress-root"
        role="progressbar"
        style="width: 56px; height: 56px; transform: rotate(-90deg);"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.5"
            stroke-width="3"
            style="stroke-dasharray: 128.805; stroke-dashoffset: 0.000px;"
          />
        </svg>
      </span>
    </div>
    <div
      class="MuiStack-root css-j3ygch-MuiStack-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
      >
        Test Title
      </h6>
      <span
        class="MuiBox-root css-10d62iu"
      >
        Balance
      </span>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
      >
        100
         MMC
      </h6>
    </div>
  </div>
</div>
`;
